baseband/vlbi_base/tests/test_vlbi_base.py

```diff
@@ -425,34 +425,147 @@ def test_vlbi_file_base(self, tmpdir):
+        assert sbs._unsqueeze(data).shape[1:] == unsliced_shape_short
+
+    @pytest.mark.parametrize(('subset', 'sliced_shape',
+                              'squeezed_shape', 'squeezed_n'),
+                             [(0, (1, 21, 33, 1, 2),
+                               (21, 33, 2), ('n1', 'n2', 'n4')),
+                              ((0, 13), (1, 1, 33, 1, 2),
+                               (33, 2), ('n2', 'n4')),
+                              ((0, np.array([2, 8, 9])[:, np.newaxis], [1, 7]),
+                               (1, 3, 2, 1, 2), (3, 2, 2), ('n1', 'n2', 'n4')),
+                              ((Ellipsis, 0, 1), (1, 21, 33, 1, 1),
+                               (21, 33), ('n1', 'n2')),
+                              ((0, slice(1, None, 4), slice(None),
+                                Ellipsis, [1]), (1, 5, 33, 1, 1),
+                               (5, 33), ('n1', 'n2')),
+                              ((0, slice(None, 1, -4)), (1, 5, 33, 1, 2),
+                               (5, 33, 2), ('n1', 'n2', 'n4'))])
+    def test_subset(self, subset, sliced_shape, squeezed_shape, squeezed_n):
+        # Tests subsetting and squeezing subset samples.
+
+        smp_shp_mkr = namedtuple('SampleShape',
+                                 'n0, n1, n2, n3, n4')
+        unsliced_shape = (1, 21, 33, 1, 2)
+        data = np.empty((100,) + unsliced_shape, dtype='float32')
+        sb = self.make_streamwriter_with_shape(smp_shp_mkr, unsliced_shape,
+                                               subset=subset, squeeze=False)
+
+        assert sb.sample_shape == sliced_shape
+        subset_data = data[(slice(0, 12),) + sb.subset]
+        assert subset_data.shape == (12,) + sb.sample_shape
+
+        # Check that squeezing subset data is self-consistent and consistent
+        # with squeezing the data from above.
+        sbs = self.make_streamwriter_with_shape(smp_shp_mkr, unsliced_shape,
+                                                subset=subset, squeeze=True)
+        assert sbs.sample_shape == squeezed_shape
+        assert sbs.sample_shape._fields == squeezed_n
+        sbs_data = data[(slice(0, 12),) + sbs.subset]
+        assert sbs_data.squeeze().shape == (12,) + sbs.sample_shape
+        assert sbs_data.squeeze().shape == subset_data.squeeze().shape
+
+    def test_subset_specialindexing(self):
+        # Tests that single integer subsets are converted to slices if
+        # squeeze is False, but any advanced indexing that reduces
+        # dimensionality returns TypeErrors.
+
+        smp_shp_mkr = namedtuple('SampleShape',
+                                 'n0, n1, n2, n3, n4')
+        unsliced_shape = (1, 21, 33, 4, 1)
+
+        # When squeeze = False, passing ints returns slices.
+        sb = self.make_streamwriter_with_shape(smp_shp_mkr, unsliced_shape,
+                                               subset=None, squeeze=False)
+        subset, subset_wrapints = sb._get_subset(0)
```

Comment:
I would generally avoid testing private functions, as those are implementation details. Just check that you get the right answers instead.

Reply:
Switched to just making new classes and attempting to subset with them.